deepxde/icbc/boundary_conditions.py

```diff
@@ -45,28 +55,57 @@ def __init__(self, geom, on_boundary, component):
             utils.return_tensor(self.geom.boundary_normal)
         )
 
-    def filter(self, X):
+    def filter(self, X: NDArray[Any]) -> NDArray[np.bool_]:
         return X[self.on_boundary(X, self.geom.on_boundary(X))]
 
-    def collocation_points(self, X):
+    def collocation_points(self, X: NDArray[Any]) -> NDArray[Any]:
         return self.filter(X)
 
-    def normal_derivative(self, X, inputs, outputs, beg, end):
+    def normal_derivative(
+        self,
+        X: NDArray[Any],
+        inputs: TensorOrTensors,
+        outputs: Tensor,
+        beg: int,
+        end: int,
+    ) -> Tensor:
         dydx = grad.jacobian(outputs, inputs, i=self.component, j=None)[beg:end]
         n = self.boundary_normal(X, beg, end, None)
         return bkd.sum(dydx * n, 1, keepdims=True)
 
     @abstractmethod
-    def error(self, X, inputs, outputs, beg, end, aux_var=None):
+    def error(
+        self,
+        X: NDArray[Any],
+        inputs: TensorOrTensors,
+        outputs: Tensor,
+        beg: int,
+        end: int,
+        aux_var: Union[NDArray[np.float_], None] = None,
```

Comment:
This should work once the suggestion above is applied.
```suggestion
        aux_var: NDArray[np.float_] | None = None,
```